xla/tests/dot_operation_test.cc

```diff
@@ -356,6 +381,16 @@ void ParametricDotTest::TestImpl() {
 
   XlaBuilder builder(TestName());
   auto prim_type = primitive_util::NativeToPrimitiveType<NativeT>();
+    
+  if(enable_grad_xy_) {
+    FrontendAttributes attributes;
+    if(grad_xy_flip_ == 0)
+      (*attributes.mutable_map())["grad_x"] = "true";
+    else
+      (*attributes.mutable_map())["grad_y"] = "true";
+    grad_xy_flip_ ^= 1;
```

Comment:
I think this is not a good idea, as it implies that test order becomes important to potentially reproduce issues. Please set the flag explicitly to 1/0 for each test.

Reply:
Yes, basically it does not matter which one of 'grad_x' or 'grad_y' is set for each test. I just wanted to test both of them. But I reimplemented it in another way, so that attributes are assigned to each tests deterministically based on test Param.